### PR TITLE
Fetch config with API token

### PR DESCRIPTION
**Pull request by phamann**
**⚠️ Note this is a PR into the v.2.0.0 branch and not master**

### TL;DR
Adds utilities to parse the API token from the script tag `src` url and pass to `Worker.init()` to fetch the config from API. 

### Why?
As we are no longer dynamically transcluding the API token into the JavaScript at the edge (thus allowing for the same script for all consumers), we need to obtain it from the script tag, i,.e. from `<script defer src="https://www.fastly-insights.com/insights.js?k=1-2-3-4"></script>`. Therefore we have to use a utility to find the tag and parse out the parameters. 

### Notes:
- Due to browser support and the size of the polyfill, we can't just use the newer URL interface: `let params = (new URL(src)).searchParams;` and have opted for writing a custom util. https://github.com/fastly/insights.js/commit/51aec8633075d9233294bb33dfe7f171a38fc155
- It seems that the the `HtmlWebpackPlugin` doesn't allow you to easily add query parameters to the chunk urls nor are there any other plugins that allow this. Therefore I had to write my own for this behaviour in https://github.com/fastly/insights.js/commit/bb7c3e316d1b57b4300852f78f12db2aae0b0ae2
- I've now added the correct API hostname and paths and therefore also added a `PRODUCTION` global via webpack `DefinePlugin`. The nice thing about this is the noop gets compiled away at compile time and therefore the final bundle only have the value for that environment.

### CODE REVIEW

**Comment by clight-fastly** [comment]
Are we sure that we need to write our own util for lack of `URL` when (I think?) the code that needs it is guarded behind a browser compat check?
https://caniuse.com/#feat=url

**Comment by phamann** [comment]
@clight-fastly I've now refactored to use `new URL()`.
